src/mtype.h

```diff
@@ -145,6 +145,7 @@ enum m_flag : int {
     MF_INTERIOR_AMMO,       // Monster contain's its ammo inside itself, no need to load on launch.
     MF_CLIMBS,              // Monsters that can climb certain terrain and furniture
     MF_PUSH_MON,            // Monsters that can push creatures out of their way
+    MF_EXPERIENCE_UPGRADE,//monsters like pokemon that don't upgrade based on time alone but experience.
```

Comment:
Should be documented in doc/JSON_FLAGS.md (the list of monster flags starts around line 177).